src/Features/LanguageServer/Protocol/Workspaces/LspWorkspaceRegistrationService.cs

```diff
@@ -43,6 +43,16 @@ public virtual void Register(Workspace workspace)
         workspace.WorkspaceChanged += OnLspWorkspaceChanged;
     }
 
+    public void Deregister(Workspace workspace)
+    {
+        lock (_gate)
+        {
+            _registrations = _registrations.Remove(workspace);
+        }
+
+        workspace.WorkspaceChanged -= OnLspWorkspaceChanged;
```

Comment:
Shouldn't this be moved first, just in case we deregister, then get an event, then unhook the event handler?

Reply:
VSMac will likely not send any events after the workspace is disposed, but yeah, that's a good suggestion.

Reply:
good call, fixed!